aten/src/ATen/SavedTensorHooks.cpp

```diff
@@ -15,6 +14,43 @@ namespace {
   static bool is_enabled(false);
 }
 
+static void assertSavedTensorHooksNotDisabled() {
+  if (!tls.is_disabled) {
+    return;
+  }
+  if (tls.disabled_error_message.has_value()) {
+    TORCH_CHECK(false, tls.disabled_error_message.value());
+  }
+  TORCH_CHECK(false, "Attempted to use Saved Tensor Hooks but they were disabled");
+}
+
+bool SavedTensorDefaultHooks::is_disabled() {
+  return tls.is_disabled;
+}
+
+void SavedTensorDefaultHooks::set_disabled(bool disabled) {
```

Comment:
Could we pass the error message directly here?
Because otherwise the check below will never actually see the error message as it gets registered later.

Reply:
> I think we should fold the flag and message into a single field so that you can never be in a state where there is an error message but it is not disabled yet.

Sure. NB: this complicates the API, because it doesn't make sense to do `set_disabled(disabled=False, "some error message")`. So instead we need a set_disabled(error_message) and a set_enabled(). If you're OK with that then I'll go ahead with it.

Reply:
Looks like in the python context manager, the error message is set first, so it would been seen.
I'd still second combining the two APIs into one though, just to avoid any extra confusion.

Reply:
Another idea could be to just have an assert that the message is empty when disabled=False?

Reply:
Non-symetric API sounds ok for me.

Reply:
Note that set_enabled(disabled=False) raising an error if you pass the optional message is good as well.

Reply:
Okay, I'll play around with it, thanks for the feedback